src/chocolatey.resources/helpers/functions/Uninstall-ChocolateyPath.ps1

```diff
@@ -0,0 +1,117 @@
+# Copyright 2011 - Present RealDimensions Software, LLC & original authors/contributors from https://github.com/chocolatey/chocolatey
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function Uninstall-ChocolateyPath {
+<#
+.SYNOPSIS
+**NOTE:** Administrative Access Required when `-PathType 'Machine'.`
+
+This puts a directory to the PATH environment variable of the
+requested scope (Machine or User).
+
+.DESCRIPTION
+Removes path from target path scope.  Removes multiple occurances (if they exist)
+and all occurances with or without a trailing slash.
+
+.NOTES
+This command will assert UAC/Admin privileges on the machine if
+`-PathType 'Machine'`.
+
+This is used when the application/tool is not being linked by Chocolatey
+(not in the lib folder).
+
+.INPUTS
+None
+
+.OUTPUTS
+None
+
+.PARAMETER PathToUninstall
+The full path to a location to remove from the PATH.
+
+.PARAMETER PathType
+Which PATH to remove it from. If specifying `Machine`, this requires admin
+privileges to run correctly.
+
+.PARAMETER IgnoredArguments
+Allows splatting with arguments that do not apply. Do not use directly.
+
+.EXAMPLE
+Uninstall-ChocolateyPath -PathToUninstall "$($env:SystemDrive)\tools\gittfs"
+
+.EXAMPLE
+Uninstall-ChocolateyPath "$($env:SystemDrive)\Program Files\MySQL\MySQL Server 5.5\bin" -PathType 'Machine'
+
+.LINK
+Install-ChocolateyPath
+
+.LINK
+Install-ChocolateyEnvironmentVariable
+
+.LINK
+Get-EnvironmentVariable
+
+.LINK
+Set-EnvironmentVariable
+
+.LINK
+Get-ToolsLocation
+#>
+param(
+  [parameter(Mandatory=$true, Position=0)][alias("Path")][string] $PathToUninstall,
+  [parameter(Mandatory=$false, Position=1)][ValidateSet('User','Machine')][System.EnvironmentVariableTarget] $pathType = [System.EnvironmentVariableTarget]::User,
+  [parameter(ValueFromRemainingArguments = $true)][Object[]] $ignoredArguments
+)
+  Write-Debug "Running 'Uninstall-ChocolateyPath' with PathToUninstall:`'$PathToUninstall`'";
+  Write-Output "Only evaluating and updating path scope `"$pathType`", path will not be assessed nor removed for other scope, so path may exist in other scope as well."
+  $originalPathToUninstall = $PathToUninstall
+  #First half on handling trailing slash properly - remove it from requested path:
+  $PathToUninstall = $PathToUninstall.trimend('\')
```

Comment:
What is trimend and is it available in PowerShell v2+ (and thus .NET Framework 2.0+)?